backend/app/services/messages.js

```diff
@@ -0,0 +1,64 @@
+const request = require('request-promise');
+const config = require('../../config');
+
+module.exports = {
+    SYSTEM_MESSAGE_USER_TOKEN_FIELD: 'SYSTEM_MESSAGE_USER_TOKEN',
+    SYSTEM_MESSAGE_SUBJECT: 'System Message',
+    createSystemMessageUser: function() {
+        const path = '/user';
+
+        const requestOptions = {
+            url: config.authService + path,
+            method: 'POST',
+            json: {
+                username: config.systemMessageUser,
+                email: config.systemMessageUser,
+                password: config.systemMessagePassword,
+                scopes: [],
+            }
+        };
+
+        return request(requestOptions)
```

Comment:
Is this the proper error handling setup? We return a promise, except we could also return or throw an error. This should be clearly documented. I would actually advise simply returning a promise and allowing the calling code to unpack error handling.

Reply:
I filtered out the 400 and returned the error as a result because that's the error you get if the user already exists (among other causes), which, for this function, is not an error. This is an example of where a semantic error code in the 400 response from the auth service would be helpful. I can either add that to auth under this PR, which is pretty sprawly, or just let the 400 go up to this function's caller, because it is equivalent and I agree this is pretty confusing